photon-ml/src/main/scala/com/linkedin/photon/ml/ModelTraining.scala

```diff
@@ -155,42 +160,34 @@ object ModelTraining extends Logging {
       s"${warmStartModels.keys.mkString(", ")}")
 
     val initWeightsAndModels = List[(Double, GeneralizedLinearModel)]()
-    val (finalOptimizationProblem, finalWeightsAndModels) = sortedRegularizationWeights
-      .foldLeft((initOptimizationProblem, initWeightsAndModels)) {
-        case ((optimizationProblem, List()), currentWeight) =>
+    val finalWeightsAndModels = sortedRegularizationWeights
+      .foldLeft(initWeightsAndModels) {
+        case (List(), currentWeight) =>
           // Initialize the list with the result from the first regularization weight
-          val updatedOptimizationProblem = optimizationProblem.updateObjective(
-            normalizationContextProvider, currentWeight)
+          optimizationProblem.updateRegularizationWeight(currentWeight)
```

Comment:
@joshvfleming 
Updating regularization weight is stateful now. I did it because it's the easy way out, though I would not object to going back through the code and removing state wherever possible.